src/cargo/ops/resolve.rs

```diff
@@ -160,12 +160,12 @@ pub fn resolve_with_previous<'a, 'cfg>(
         );
     }
 
-    let ref keep = |p: &&'a PackageId| {
+    let keep = &(|p: &&'a PackageId| {
```

Comment:
this seems unfortunate, can you open an issue about it in the clippy repo?

Reply:
I think `&(|| { ... })` is not very good. the `ref` seemed a lot prettier. We should not report removing the `ref` in favour of a `&` if the rhs is a closure.